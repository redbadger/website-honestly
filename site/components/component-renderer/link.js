// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
