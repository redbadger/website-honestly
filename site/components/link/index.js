// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
