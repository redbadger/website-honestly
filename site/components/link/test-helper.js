// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
