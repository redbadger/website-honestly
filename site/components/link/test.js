// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
