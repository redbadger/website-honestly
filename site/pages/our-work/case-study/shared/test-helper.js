// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
