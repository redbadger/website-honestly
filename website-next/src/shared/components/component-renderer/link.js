import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
