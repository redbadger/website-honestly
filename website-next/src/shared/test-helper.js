import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
