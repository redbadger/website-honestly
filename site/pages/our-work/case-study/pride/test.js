import React from 'react';
import { mount } from 'enzyme';

import { Context } from '../../../../components/link/test-helper';
import MockedWhatToReadNext from '../shared/test-helper';
import Pride from './index';

jest.mock('../shared/what-to-read-next', () => MockedWhatToReadNext);

describe('site/pages/our-work/case-study/pride/', () => {
  it('renders correctly', () => {
    expect(
      mount(
        <Context>
          <Pride />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
