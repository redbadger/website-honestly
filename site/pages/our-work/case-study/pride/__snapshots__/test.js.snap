// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/our-work/case-study/pride/ renders correctly 1`] = `
Array [
  <div
    class="headerImgWrapper"
  >
    <picture>
      <source
        class="headerImg"
        srcset="header.gif"
      />
      <img
        alt=""
        class="headerImg"
        src="header.gif"
      />
    </picture>
  </div>,
  <div
    class="body"
  >
    <div
      class="heading"
    >
      <h1
        class="fontL "
      >
        A labour of love – delivering a flagship, best-in-class digital experience for the Pride in London community
      </h1>
    </div>
    <div
      class="section"
    >
      <div
        class="sectionBody"
      >
        <div
          class="quotation quote"
        >
          <div
            class="quotation__text"
          >
            <blockquote
              class="quotation__quote"
            >
              I loved working with Red Badger. From the very start of the project I felt the passion and energy from each and every team member. You proved yourself to be a well oiled machine getting some really awesome, creative work out and working to a schedule and deadlines. Everyone at Pride is super excited and happy with what we’ve done.
            </blockquote>
          </div>
          <div
            class="quotation__author"
          >
            <div
              class="avatar__container"
              style="width:70px;height:70px;min-width:70px;min-height:70px"
            >
              <img
                alt="Kristof Hamilton"
                src="Kristof_HS.jpg"
              />
            </div>
            <div
              class="quotation__author__wrapper"
            >
              <div
                class="quotation__author__name"
              >
                — Kristof Hamilton
              </div>
              <div
                class="quotation__author__title"
              >
                Product Owner at Pride in London
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="section"
    >
      <h2
        class="sectionHeading"
      >
        <span
          class="redTitle"
        >
          Let's make things better
        </span>
        Encouraging festival discovery with a mobile app
      </h2>
      <div
        class="sectionBody"
      >
        <p
          class="fontXS paragraph"
        >
          Pride in London is a non-profit, volunteer-led organisation which works to provide a platform for every part of London’s LGBTQ+ community and campaigns for freedoms that will allow them to live their lives on a genuinely equal footing. It’s ​the ​world’s largest curated ​LGBTQ+ ​festival ​with ​over ​300 ​LGBTQ+ events, ​culminating ​in a Pride parade ​through ​Central ​London. Over 1 million visitors join the day, including 30,000 parade participants from ​all ​over ​the ​UK ​and abroad.
        </p>
        <p
          class="fontXS paragraph"
        >
          Pride In London released their first app for the 2017 parade. 2018 saw more ambitious goals for the events and also the digital experience. Red Badger worked with Pride in London to define the vision for the project. As part of this we mapped a set of key requirements, at the core of which were inclusivity and accessibility, reflecting values and behaviour of Pride in London and respecting the diversity of the community they serve. The main requirement of the app was to encourage the discovery of the events and help Pride goers navigate the parade. Due to organisational challenges, we were not able to reuse the code developed for the previous app thus the product had to be built from scratch.
        </p>
        <div>
          <div
            class="wrapper"
          >
            <picture>
              <img
                alt="Phone with the Pride app on in screen displaying a map, text reads: 'Easily find bars and events on the day'"
                class="leftImg"
                src="left.jpg"
              />
            </picture>
            <picture>
              <img
                alt="Phone with the Pride app on in screen displaying a list of events, text reads: 'Plan your perfect festival with ease and love'"
                class="middleImg"
                src="middle.jpg"
              />
            </picture>
            <picture>
              <img
                alt="Phone with the Pride app on in screen displaying a list of event categories, text reads: 'Find diverse experiences from cabaret to films'"
                class="rightImg"
                src="right.jpg"
              />
            </picture>
          </div>
          <div>
            <form
              action=""
              class="controls"
            >
              <label>
                <span
                  class="fakeControl "
                />
                <span
                  class="controlsLabel"
                >
                  First
                </span>
                <input
                  class="controlsInput"
                  name="image"
                  type="radio"
                  value="0"
                />
              </label>
              <label>
                <span
                  class="fakeControl fakeControlActive"
                />
                <span
                  class="controlsLabel"
                >
                  First
                </span>
                <input
                  class="controlsInput"
                  name="image"
                  type="radio"
                  value="1"
                />
              </label>
              <label>
                <span
                  class="fakeControl "
                />
                <span
                  class="controlsLabel"
                >
                  First
                </span>
                <input
                  class="controlsInput"
                  name="image"
                  type="radio"
                  value="2"
                />
              </label>
            </form>
          </div>
        </div>
        <h3
          class="fontS2 "
        >
          The right project for Red Badger
        </h3>
        <p
          class="fontXS paragraph"
        >
          As a business we've always been proud of the work we do and two years ago we decided to focus more on delivering social good, leading to us establishing a Social Value team. The Pride in London app was the perfect project to demonstrate what we could achieve by volunteering our time.
        </p>
        <p
          class="fontXS paragraph"
        >
          It was the right project for us on many fronts, allowing us to:
        </p>
        <ul
          class="list"
        >
          <li>
            Work with a brand that chimes with Red Badger’s own values of valuing diversity, and being inclusive and accessible
          </li>
          <li>
            Design and deliver the app as a truly cross-functional team 
          </li>
          <li>
            Choose the right tech for the job 
          </li>
          <li>
            Demonstrate how best in class delivery can create social value
          </li>
        </ul>
      </div>
    </div>
    <div
      class="section"
    >
      <h2
        class="sectionHeading"
      >
        <span
          class="redTitle"
        >
          Do the right thing. Do the thing right
        </span>
        Working as a cross-functional volunteer team
      </h2>
      <div
        class="sectionBody"
      >
        <h3
          class="fontS2 "
        >
          Remote cross-functional team
        </h3>
        <p
          class="fontXS paragraph"
        >
          At Red Badger we usually work in cross-functional teams, co-locating with our clients to enable short feedback loops and highly efficient teams. In a voluntary project this wasn’t possible and it was particularly challenging.
        </p>
        <p
          class="fontXS paragraph"
        >
          At the beginning of the project we had volunteers from our Strategy, Branding, User Experience, Design, Delivery, Engineering teams and Pride in London, in the same room, collaborating on the product vision and roadmap. We agreed that the app had to be a place to promote and showcase the diversity of the 2018 festival and parade, help people discover, plan and get involved in events, while also making it simple and easy to use.
        </p>
        <p
          class="fontXS paragraph"
        >
          We ran a branding workshop early on. This helped define the design approach and the creative vision for the app. Our approach centred on a richer, more cohesive experience for the user.
        </p>
        <img
          alt=""
          class="articleImg"
          src="collaboration.jpg"
        />
        <h3
          class="fontS2 "
        >
          Choosing React Native as the right tech for the job
        </h3>
        <p
          class="fontXS paragraph"
        >
          When looking into which tech would be best to deliver a fully functioning app across two platforms, React Native came up top. Firstly, it meant we could open source the project and give back to the tech community which we collectively rely so heavily on. Secondly, it meant that we could develop our skills and expertise in a growing technology that has incredible potential for the future – not just for Pride in London, but for our other clients too. Thirdly, given tight timelines, we could deliver a minimum desirable mobile product for both iOS and Android platforms quickly, and iterate from there.
        </p>
      </div>
    </div>
    <div
      class="section"
    >
      <h2
        class="sectionHeading"
      >
        <span
          class="redTitle"
        >
          Creating lasting change
        </span>
        Proud to have built the Pride in London Mobile App
      </h2>
      <div
        class="sectionBody"
      >
        <p
          class="fontXS paragraph"
        >
          Proud is the word that best encompasses the ethos of the project. We’re incredibly proud of the quality of work and the product we delivered for Pride In London 2018 festival.
        </p>
        <p
          class="fontXS paragraph"
        >
          Amongst the things we’re most proud of are:
        </p>
        <ul
          class="list"
        >
          <li>
            Partnering with an organisation that promotes equality, diversity and inclusion
          </li>
          <li>
            Developing an open source project that can live beyond our involvement and be easily improved for the next year’s events
          </li>
          <li>
            Delivering a 5★ app experience with AA accessibility standards 
          </li>
          <li>
            Helping 20,000+ people navigate the festival and parade
          </li>
          <li>
            Delivering an end-to-end mobile product in 6 months working as a genuinely cross-functional team
          </li>
          <li>
            Delivering a project using React Native and learning some valuable lessons along the way
          </li>
        </ul>
        <p
          class="fontXS paragraph"
        >
          You can read more in-depth about 
          <a
            href="https://blog.red-badger.com/2018/6/27/branding-the-pride-in-london-app"
          >
            Branding
          </a>
          , 
          <a
            href="https://blog.red-badger.com/2018/6/28/designing-pride-into-the-experience"
          >
            UX
          </a>
          , 
          <a
            href="https://red-badger.com/blog/2018/6/28/pride-in-london-going-native"
          >
            React Native
          </a>
          , 
          <a
            href="https://blog.red-badger.com/blog/2018/6/28/open-source-with-pride"
          >
            Open Source
          </a>
           and 
          <a
            href="https://blog.red-badger.com/2018/6/28/the-pride-in-london-2018-app-delivering-a-rainbow"
          >
            Delivery
          </a>
          .
        </p>
        <p
          class="fontXS paragraph"
        >
          Since completing this project we have been approached and engaged with some new clients on mobile projects. If you have a mobile app project in mind we’d love to talk to you about how to apply our knowledge and skills to help you deliver great mobile experiences for your customers. Please get in touch on 
          <a
            class="link"
            href="email:hello@red-badger.com"
          >
            hello@red-badger.com
          </a>
          .
        </p>
        <p
          class="fontXS paragraph"
        >
          Are you a charity or NGO looking for help with your project? Our social value team offer pro bono or reduced rates. Please contact us at 
          <a
            class="link"
            href="email:socialvalue@red-badger.com"
          >
            socialvalue@red-badger.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="section"
    >
      <div
        class="videoContainer"
      >
        <span>
          <div
            id="pride-video"
          />
        </span>
      </div>
    </div>
    <div
      class="section"
    >
      <div
        class="sectionBody"
      >
        <div
          class="Card"
        >
          <div
            class="listBox listBox"
          >
            <h3
              class="listBox__heading"
            >
              Results
            </h3>
            <ul>
              <li
                class="listBox__element listBox__element--results listBox__item"
              >
                <div>
                  21,000+ app downloads, +50% vs. 2017
                </div>
                <div />
              </li>
              <li
                class="listBox__element listBox__element--results listBox__item"
              >
                <div>
                  1,000 daily active users, +280% vs. 2017
                </div>
                <div />
              </li>
              <li
                class="listBox__element listBox__element--results listBox__item"
              >
                <div>
                  1000+ hours on the app on Parade Day alone
                </div>
                <div />
              </li>
              <li
                class="listBox__element listBox__element--results listBox__item"
              >
                <div>
                  5★ user ratings on iOS and Android platforms
                </div>
                <div />
              </li>
              <li
                class="listBox__element listBox__element--results listBox__item"
              >
                <div>
                  Featured as an app of the day on App Store
                </div>
                <div />
              </li>
              <li
                class="listBox__element listBox__element--results listBox__item"
              >
                <div>
                  ‘AA’ accessibility standard
                </div>
                <div />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="section"
    >
      <div
        class="sectionBody"
      >
        <div
          class="quotation quote"
        >
          <div
            class="quotation__text"
          >
            <blockquote
              class="quotation__quote"
            >
              Pride in London’s new app was an exciting challenge for Red Badger and all the volunteers involved. As a pro-bono, voluntary, open source project that has been built for the community, by the community, it’s been a true labour of love and collaboration. The app is a culmination of all the volunteers’ work and truly reflects the enthusiasm they brought to the project.
            </blockquote>
          </div>
          <div
            class="quotation__author"
          >
            <div
              class="avatar__container"
              style="width:70px;height:70px;min-width:70px;min-height:70px"
            >
              <img
                alt="Lani Shamash"
                src="Lani_HS.jpg"
              />
            </div>
            <div
              class="quotation__author__wrapper"
            >
              <div
                class="quotation__author__name"
              >
                — Lani Shamash
              </div>
              <div
                class="quotation__author__title"
              >
                Delivery Director at Red Badger
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="whatNext"
  >
    <div
      class="whatNext__tilesContainer"
    >
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/pride/"
          >
            <img
              alt="Pride in London"
              class="whatNext__image"
              src="pride.gif"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Pride in London
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/pride/"
        >
          <h2
            class="whatNext__title"
          >
            Delivering a best-in-class digital experience
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/financial-times/"
          >
            <img
              alt="Financial Times"
              class="whatNext__image"
              src="financialTimesReadMore.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Financial Times
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/financial-times/"
        >
          <h2
            class="whatNext__title"
          >
            Lasting change for a media giant
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/car-trawler/"
          >
            <img
              alt="Car Trawler"
              class="whatNext__image"
              src="carTrawler.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Car Trawler
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/car-trawler/"
        >
          <h2
            class="whatNext__title"
          >
            Creating complete CMS control for travel brands
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
    </div>
    <a
      class="whatNext__button"
      href="/our-work/"
    >
      See more of our work
    </a>
  </div>,
  <section
    class="contactUsContainer"
    data-cy="contact-us-slice"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Create & validate new ideas
        </li>
        <li
          class="item"
        >
          Deliver great quality products & services, fast
        </li>
        <li
          class="item"
        >
          Be bold with technology
        </li>
        <li
          class="item"
        >
          Be more customer centric
        </li>
        <li
          class="item"
        >
          Improve efficiency with lean practices
        </li>
        <li
          class="item"
        >
          Build capability & confidence
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        data-cy="email-link"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>,
]
`;
